postgres-appliance/build.sh

```diff
@@ -24,11 +38,34 @@ cat > scm-source.json <<__EOT__
 }
 __EOT__
 
-${DOCKERCMD} $@
-EXITCODE=$?
+function run_or_fail() {
+    "$@"
+    local EXITCODE=$?
+    if  [[ $EXITCODE != 0 ]]; then
+        echo "'$@' failed with exitcode $EXITCODE"
+        exit $EXITCODE
+    fi
+}
+
+readonly OLD_BUILD_ID=$(docker images -q $IMGNAME-build)
+
+function squash_new_image() {
+    local NEW_BUILD_ID=$(docker images -q $IMGNAME-build)
+    local TAG_OF=$(docker images --format "{{.ID}} {{.Repository}}:{{.Tag}}" \
+            | grep "^$NEW_BUILD_ID " | grep -v "^$NEW_BUILD_ID $IMGNAME-build" \
+            | awk '{print $2; exit 0}')
+
+    # new "-build" image has the same id as already exiting one
+    [[ ! -z $TAG_OF ]] && docker tag ${TAG_OF%-build}-squashed $IMGNAME-squashed && return 0
+
+    [[ "$OLD_BUILD_ID" != "$NEW_BUILD_ID" || -z "$(docker images -q $IMGNAME-squashed)" ]] \
+            && run_or_fail docker-squash -t $IMGNAME-squashed $IMGNAME-build
+}
+
+run_or_fail docker build "${build_args[@]}" -f Dockerfile.build
```

Comment:
I think this is missing the trailing '.' as the directory where the Docker should operate (also on line 69). At least it didn't work on my system without it.

Reply:
This build script accepts the same arguments as "docker build".
Basically it is a replacement for "docker build".
I invoke it as: `./build.sh --build-arg COMPRESS=true -t my-cool-image:my-cool-tag .`